src/box/lua/key_def.c

```diff
@@ -124,12 +124,15 @@ luaT_key_def_set_part(struct lua_State *L, struct key_part_def *part,
 	/* Set part->type. */
 	lua_pushstring(L, "type");
 	lua_gettable(L, -2);
-	if (lua_isnil(L, -1)) {
+	size_t type_len;
+	const char *type_name;
+	bool type_is_nil = lua_isnil(L, -1);
+	if (!type_is_nil)
+		type_name = lua_tolstring(L, -1, &type_len);
+	if (type_is_nil || type_len == 0) {
```

Comment:
This results in confusing behavior:

```
tarantool> require('key_def').new({{field = 1, type = ''}})
---
- error: type must not be nil
...

tarantool> require('key_def').new({{field = 1, type = true}})
---
- error: type must not be nil
...
```

I think that we should explicitly check that the type is a string here and fix the error message.

Reply:
I suggest to use `lua_typename(L, lua_type(L, idx))` to generate an error message with the actual Lua type. Example:

https://github.com/tarantool/tarantool/blob/6e45042473fb973c7b1ad87fcf04c9081a4c40fa/src/lua/popen.c#L618-L631

Reply:
Done